tests/optimize/test_optimize_reports.py

```diff
@@ -201,6 +202,27 @@ def test_store_backtest_stats(testdatadir, mocker):
     assert str(dump_mock.call_args_list[0][0][0]).startswith(str(testdatadir / 'testresult'))
 
 
+def test_store_backtest_candles(testdatadir, mocker):
+
+    dump_mock = mocker.patch('freqtrade.optimize.optimize_reports.file_dump_pickle')
```

Comment:
this test seems fine at first glance.

i'd however also add a 2nd test which uses `tmpdir` instead of `testdatadir` - and which doesn't mock file-writing.
then you can test "does it really write a file" - and also "can i read the written file" - which is quite important :laughing: 

That's however slightly complicated by the fact that freqtrade doesn't have a method to actually read this file back - so that part can't really sanely be tested for now (see other comment on that).

Reply:
OK i'll code in an actual write. The buy_reasons script is the reading part, which I can also add (see below)

Reply:
Test implemented in [b3cb722](https://github.com/freqtrade/freqtrade/pull/6543/commits/b3cb7226467c359b0748cb4866f8f8fcfa649a53)